src/Core/Extensible.php

```diff
@@ -245,9 +245,9 @@ public static function remove_extension($extension)
     {
         $class = get_called_class();
 
-        // Build filtered extension list
+        // Build extension list
         $found = false;
-        $config = Config::inst()->get($class, 'extensions', Config::EXCLUDE_EXTRA_SOURCES | Config::UNINHERITED) ?: [];
+        $config = Config::inst()->get($class, 'extensions') ?: [];
```

Comment:
If you don't have EXCLUDE_EXTRA_SOURCES it'll call extensions when calculating the extensions. Seems risky.

Reply:
Instead of removing UNINHERITED isn't it worth finding out the specific class the extension is added to?

Reply:
We can add EXCLUDE_EXTRA_SOURCES back in again no problem I think.

This logic was actually fine the way it was, it's the fact that what comes out of it is the canonical list that gets set back to the config for the class, excluding parent extensions that were there originally. Perhaps the parent extensions should never have been there, and `has_extension()` should factor in parent class config when it checks for them